modules/exploits/linux/ssh/vyos_restricted_shell_privesc.rb

```diff
@@ -0,0 +1,222 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'net/ssh'
+require 'net/ssh/command_stream'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GreatRanking
+
+  include Msf::Exploit::Remote::SSH
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'VyOS restricted-shell Escape and Privilege Escalation',
+        'Description' => %q{
+          This module exploits command injection vulnerabilities and an insecure
+          default sudo configuration on VyOS versions 1.0.0 <= 1.1.8 to execute
+          arbitrary system commands as root.
+
+          VyOS features a `restricted-shell` system shell intended for use by
+          low privilege users with operator privileges. This module exploits
+          a vulnerability in the `telnet` command to break out of the restricted
+          shell, then uses sudo to exploit a command injection vulnerability in
+          `/opt/vyatta/bin/sudo-users/vyatta-show-lldp.pl` to execute commands
+          with root privileges.
+
+          This module has been tested successfully on VyOS 1.1.8 amd64 and
+          VyOS 1.0.0 i386.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Rich Mirch', # discovery and exploit
+          'bcoles' # metasploit
+        ],
+        'References' =>
+        [
+          [ 'CVE', '2018-18556' ],
+          [ 'URL', 'https://blog.vyos.io/the-operator-level-is-proved-insecure-and-will-be-removed-in-the-next-releases' ],
+          [ 'URL', 'https://blog.mirch.io/2018/11/05/cve-2018-18556-vyos-privilege-escalation-via-sudo-pppd-for-operator-users/' ],
```

Comment:
Just a note that this site is currently giving Error 526 Invalid SSL certificate